### PR TITLE
Add helpers to generate struct insert/update SQL

### PR DESCRIPTION
**Pull request by hashmap**

### CODE REVIEW

File: pg/generators.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	"reflect"
+	"strings"
+	"unicode"
+)
+
+// GenerateInsertSQL generates INSERT statemenet for a given instance of some structure
+// and a list of values to be used in sqlx get/exec
+// table - name of DB table where struct will be stored
+// o - instance to be stored
+// it returns sql statemenet and list of values
```
Comment (PragmaticCypher) [inline pg/generators.go]: *statement

File: pg/generators_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		FirstName: "Max",
+		LastName:  "Mustermann",
+	},
+}
+
+func TestGenerateInsertSQL(t *testing.T) {
+	sql, vals := GenerateInsertSQL("users", user)
+	expectedSql := "INSERT INTO users (id, email, created_at, first_name, last_name) VALUES ($1, $2, $3, $4, $5) RETURNING *"
+	expectedVals := []interface{}{1, "x@optiopay.com", currentTime, "Max", "Mustermann"}
+	fmt.Printf("%s : %v", sql, vals)
```
Comment (PragmaticCypher) [inline pg/generators_test.go]: should it not be t.Logf?
https://golang.org/pkg/testing/#T.Logf

File: pg/pg.go [hunk 3]
```diff
@@ -73,6 +73,11 @@ func DB(ctx context.Context) Database {
 	return db.(Database)
 }
 
+// NewDB return database instance for a conection
```
Comment (PragmaticCypher) [inline pg/pg.go]: *connection

File: pg/generators_test.go (hunk 2 above)
Comment (hashmap) [inline pg/generators_test.go]: We don't need it at all, will remove, thanks

File: pg/generators.go (hunk 1 above)
Comment (hashmap) [inline pg/generators.go]: Fixed

File: pg/pg.go (hunk 3 above)
Comment (hashmap) [inline pg/pg.go]: Fixed

File: pg/pg.go [hunk 4]
```diff
@@ -64,6 +64,10 @@ func WithDB(ctx context.Context, db *sql.DB) context.Context {
 	return context.WithValue(ctx, "pg:db", &sqlxdb{dbx})
 }
 
+func FromSQLX(dbx *sqlx.DB) Database {
+	return &sqlxdb{dbx}
+}
+
```
Comment (PragmaticCypher) [inline pg/pg.go]: suggestion: what do you think about this naming convention 'NewFromX'

https://github.com/shopspring/decimal/blob/master/decimal.go#L147

Comment (hashmap) [inline pg/pg.go]: Sounds good

**Pull request merged**